core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -323,13 +325,43 @@ class KafkaServer(
 
         observer = Observer(config)
 
+        def initializeMemoryPoolStats(): Optional[MemoryPoolStatsStore] = {
+          if (!config.memoryPoolStatsLoggingEnable) {
+            return Optional.empty()
+          }
+
+          info(s"Memory pool stats logging is enabled, segments = " +
+            s"${config.memoryPoolStatsNumSegments}, max size = ${config.memoryPoolStatsMaxSize}, " +
+            s"logging frequency in minutes = ${config.memoryPoolStatsLoggingFrequencyMinutes}")
+          val memoryPoolStatsStore = new MemoryPoolStatsStore(config.memoryPoolStatsNumSegments, config.memoryPoolStatsMaxSize)
+          val requestStatsLogger = new MemoryPoolStatsLogger()
+
+          def publishHistogramToLog(): Unit = {
+            info("Publishing memory pool stats")
+            requestStatsLogger.logStats(memoryPoolStatsStore)
+            memoryPoolStatsStore.clear()
+          }
+
+          val histogramPublisher = new KafkaScheduler(threads = 1, "histogram-publisher-")
```

Comment:
Can we reuse the same scheduler 
https://github.com/linkedin/kafka/blob/c57acec393ba5408d1f76c7e164e6745273fbf0f/core/src/main/scala/kafka/server/KafkaServer.scala#L160

since the startup and shutting down of that scheduler is already taken care of.

Reply:
Yeah we can do that. I was not sure myself, because we use this `kafkaScheduler` in some places inside `KafkaServer`, while in other places we allocate a new `KafkaScheduler` for other purposes. So I was confused as to which way to go.

I have fixed it.